pdns/dnsdist-lua-bindings.cc

```diff
@@ -236,6 +236,19 @@ void setupLuaBindings(bool client)
                   cache->expungeByName(dname, qtype ? *qtype : QType::ANY, suffixMatch ? *suffixMatch : false);
                 }
     });
+  g_lua.registerFunction<void(std::shared_ptr<DNSDistPacketCache>::*)(const std::vector<std::pair<int,DNSName>> dnames, boost::optional<uint16_t> qtype, boost::optional<bool> suffixMatch)>("expungeByName", [](
```

Comment:
IIRC correctly this won't work, polymorphic functions needs to be implemented using `boost::variant`.